src/components/Common/RelativeDateTooltip.tsx

```diff
@@ -0,0 +1,49 @@
+import { format, isValid } from "date-fns";
+import { useTranslation } from "react-i18next";
+
+import { TooltipComponent } from "@/components/ui/tooltip";
+
+import { relativeDate } from "@/Utils/utils";
+
+type RelativeDateTooltipProps = {
+  date: Date | string;
+  className?: string;
+};
+
+export default function RelativeDateTooltip({
+  date,
+  className,
+}: RelativeDateTooltipProps) {
+  const { t } = useTranslation();
+  if (!date) return null;
+
+  const dateObj = typeof date === "string" ? new Date(date) : date;
+
+  if (!isValid(dateObj)) {
+    return (
+      <TooltipComponent content={t("invalid_date")}>
+        <span className={className}>{t("invalid_date")}</span>
+      </TooltipComponent>
+    );
+  }
+
+  const hasTime = !!(
+    dateObj.getHours() ||
+    dateObj.getMinutes() ||
+    dateObj.getSeconds()
+  );
+
+  const tooltipContent = hasTime
+    ? format(dateObj, "PPp zzz")
+    : format(dateObj, "PP zzz");
+
+  const datetimeAttr = dateObj.toISOString();
+
+  return (
+    <TooltipComponent content={tooltipContent}>
```

Comment:
1. timezone doesn't make sense when showing date alone! timezones are applicable only for timestamps.

2. keep it inlined and simplify the expression

```suggestion
  const tooltipContent = hasTime
    ? format(dateObj, "PPp zzz")
    : format(dateObj, "PP zzz");

  const datetimeAttr = dateObj.toISOString();

  return (
    <TooltipComponent content={format(date, hasTime ? "PPp zzz" : "PP"}>
```